src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -293,6 +295,68 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     pipeline(Get(FireCloudConfig.GoogleCloud.priceListUrl))
   }
 
+  /**
+    * Saves a google drive spreadsheet with the provided SpreadsheetProperties object.
+    * Note that this does not *populate* the spreadsheet, just creates it with the provided properties
+    * Call GoogleServicesDAO#updateSpreadsheet to add data once a spreadsheet is created.
+    *
+    * @param requestContext RequestContext
+    * @param userInfo       UserInfo
+    * @param props          SpreadsheetProperties
+    * @return               JsObject representing the Google Create response
+    *
+    */
+  def createSpreadsheet(requestContext: RequestContext, userInfo: UserInfo, props: SpreadsheetProperties)
+    (implicit actorRefFactory: ActorRefFactory, executionContext: ExecutionContext): Future[JsObject] = {
+
+    getUserProfile(requestContext) map { userResponse =>
```

Comment:
is this call to `getUserProfile` necessary? If the user made it this far, past our auth proxy, can't you assume they have a google account?

Reply:
We need to build a new credential object to make the spreadsheet API call. To do that, we need the bearer token. I didn't see any other way to get that other than going through the google API for that.

Reply:
But you're building the credential from the `userInfo` that is passed in:
```
val credential = new GoogleCredential().setAccessToken(userInfo.accessToken.token)
```

The `userResponse` is only used to make the `responseStr`, and `responseStr` is only used for logging.

Reply:
That will simplify things, yes, thank you!